include/nbl/core/alloc/resource_owning_vector.h

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2019-2024 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine" and was originally part of the "Irrlicht Engine"
+// For conditions of distribution and use, see copyright notice in nabla.h
+// See the original file in irrlicht source for authors
+
+#ifndef _NBL_CORE_ALLOC_RESOURCE_OWNING_VECTOR_INCLUDED_
+#define _NBL_CORE_ALLOC_RESOURCE_OWNING_VECTOR_INCLUDED_
+
+#include <memory_resource>
+
+using namespace nbl;
+
+namespace nbl::core
+{
+
+class resource_owning_vector : public std::pmr::memory_resource {
+public:
```

Comment:
our coding convention is that `{` goes on new line, and there's an extra indent before `public/protected/private` and inside those  scopes, see other code

Also we use CamelCase, this should be called `CSomething`, not sure Resource Owning Vector is a good name, try VectorViewNullMemoryResource